File: trunk/TVRename/TVRename.cpp
// TVRename.cpp : main project file.

#include "stdafx.h"
#include "UI.h"

#ifndef _DEBUG
#include "ShowException.h"
#endif

//#define NOMUTEX

using namespace TVRename;
using namespace System::Threading;
using namespace System::Security::AccessControl;
using namespace System::Security::Permissions;


[STAThreadAttribute]
int main(array<System::String ^> ^args)
{
	// Enabling Windows XP visual effects before any controls are created
	Application::EnableVisualStyles();
	Application::SetCompatibleTextRenderingDefault(false); 

#ifndef NOMUTEX
    // see if we're already running
    String ^mutexName = "TVRenameMutex";

    System::Threading::Mutex ^mutex = nullptr;

    bool requestInitialOwnership = true;
    bool createdNew = false;
    mutex = gcnew System::Threading::Mutex(requestInitialOwnership, mutexName, createdNew);
    if (!createdNew)
    {
      // we're already running
     return 0;
    }
#endif

	// Create the main window and run it

#ifndef _DEBUG
	try 
    {
#endif
      Application::Run(gcnew UI());
#ifndef NOMUTEX
	  GC::KeepAlive(mutex);
#endif

#ifndef _DEBUG
	} 
    catch (Exception ^e)
    {
      ShowException ^se = gcnew ShowException(e);
      se->ShowDialog();
    }
#endif


    return 0;
}

File: experimental/C#/tvr/TVRename.cs
//
// Main website for TVRename is http://tvrename.com
//
// Source code available at http://code.google.com/p/tvrename/
//
// This code is released under GPLv3 http://www.gnu.org/licenses/gpl.html
//

using TVRename;
using System.Threading;
using System.Security.AccessControl;
using System.Security.Permissions;
using System;
using System.Windows.Forms;

public static class GlobalMembersTVRename
{


    [STAThread]
    static int Main(string[] args)
    {
        // Enabling Windows XP visual effects before any controls are created
        Application.EnableVisualStyles();
        Application.SetCompatibleTextRenderingDefault(false);

        // see if we're already running
        string mutexName = "TVRenameMutex";

        System.Threading.Mutex mutex = null;

        bool requestInitialOwnership = true;
        bool createdNew = false;
        mutex = new System.Threading.Mutex(requestInitialOwnership, mutexName, out createdNew);
        if (!createdNew)
        {
            // we're already running
            return 0;
        }

        // Create the main window and run it

#if !DEBUG
		try
		{
#endif

        Application.Run(new UI(args));

        GC.KeepAlive(mutex);

#if !DEBUG
		}
		catch (Exception e)
		{
		  ShowException se = new ShowException(e);
		  se.ShowDialog();
		}
#endif


        return 0;
    }
}
